python_modules/dagster/dagster/_grpc/types.py

```diff
@@ -405,18 +405,23 @@ class PartitionArgs(
         "_PartitionArgs",
         [
             ("repository_origin", RemoteRepositoryOrigin),
+            ("job_name", str),
+            # this is here for backcompat. it's expected to always be f"{job_name}_partition_set"
             ("partition_set_name", str),
             ("partition_name", str),
             ("instance_ref", Optional[InstanceRef]),
+            ("selected_asset_keys", Optional[AbstractSet[AssetKey]]),
```

Comment:
I would emphasize in comments as well which arguments might not be present on older versions - some serdes objects will include a little changelog in comments at the top that you could emulate